inc/inventory/request.class.php

```diff
@@ -382,10 +500,19 @@ public function inventory($data) {
 
       if ($this->inventory->inError()) {
          foreach ($this->inventory->getErrors() as $error) {
-            $this->addError($error);
+            $this->addError($error, 500);
          }
       } else {
-          $this->addToResponse(['RESPONSE' => 'SEND']);
+         if ($this->headers->hasHeader('GLPI-Agent-ID')) {
```

Comment:
I do not understand why it should be done before, all seems good on that point.
The mode can be set from outside any HTTP request (or it may be not set at all) , relying on required header is a better solution.